rust/foxglove/src/websocket/server.rs

```diff
@@ -44,6 +44,8 @@ pub(crate) struct ServerOptions {
     pub runtime: Option<Handle>,
     pub fetch_asset_handler: Option<Box<dyn AssetHandler>>,
     pub tls_identity: Option<TlsIdentity>,
+    pub on_client_connect: Option<Box<dyn Fn() + Send + Sync>>,
```

Comment:
These should be implemented as functions in the `ServerListener` trait, not added directly to `ServerOptions`. It seems you're already doing that, so no need to add these here.